2013-05-23	Nicolas Setton  <setton@adacore.com>

	* gtk/gtkclipboard-quartz.c: Replace undefined symbol.

2013-07-23      Jerome Lambourg <lambourg@adacore.com>

        * gtk/gtkrange.c, gtk/gtkscrolledwindow.c: Better handle smooth scrolls.

--- ../gtk+-3.8.2-orig/gtk/gtkclipboard-quartz.c	2013-05-13 12:15:07.000000000 +0200
+++ gtk/gtkclipboard-quartz.c	2013-05-23 14:40:04.819631640 +0200
@@ -108,7 +108,7 @@
   memset (&selection_data, 0, sizeof (GtkSelectionData));

   selection_data.selection = clipboard->selection;
-  selection_data.target = _gtk_quartz_pasteboard_type_to_atom (type);
+  selection_data.target = gdk_quartz_pasteboard_type_to_atom_libgtk_only (type);
   selection_data.display = gdk_display_get_default ();
   selection_data.length = -1;

--- gdk/quartz/gdkwindow-quartz.c.old	2012-08-06 15:33:18.000000000 +0200
+++ gdk/quartz/gdkwindow-quartz.c	2012-08-06 15:40:51.000000000 +0200
@@ -861,10 +861,11 @@
     {
       NSRect rect = [[screens objectAtIndex:i] frame];

+      screen = [screens objectAtIndex:i];
+
       if (x >= rect.origin.x && x <= rect.origin.x + rect.size.width &&
           y >= rect.origin.y && y <= rect.origin.y + rect.size.height)
         {
-          screen = [screens objectAtIndex:i];
           break;
         }
     }
--- gtk/gtkrange.c.old	2013-07-22 13:26:18.000000000 +0200
+++ gtk/gtkrange.c	2013-07-22 14:19:21.000000000 +0200
@@ -2842,7 +2842,9 @@
   page_size = gtk_adjustment_get_page_size (adjustment);
   page_increment = gtk_adjustment_get_page_increment (adjustment);

-  if (GTK_IS_SCROLLBAR (range))
+  if (event->direction == GDK_SCROLL_SMOOTH)
+    scroll_unit = 1.0;
+  else if (GTK_IS_SCROLLBAR (range))
     scroll_unit = pow (page_size, 2.0 / 3.0);
   else
     scroll_unit = page_increment;
--- gtk/gtkscrolledwindow.c.old	2013-07-22 13:26:18.000000000 +0200
+++ gtk/gtkscrolledwindow.c	2013-07-22 14:19:56.000000000 +0200
@@ -2279,7 +2279,10 @@

           adj = gtk_range_get_adjustment (GTK_RANGE (priv->hscrollbar));
           page_size = gtk_adjustment_get_page_size (adj);
-          scroll_unit = pow (page_size, 2.0 / 3.0);
+          if (event->direction == GDK_SCROLL_SMOOTH)
+            scroll_unit = 1.0;
+          else
+            scroll_unit = pow (page_size, 2.0 / 3.0);

           new_value = CLAMP (gtk_adjustment_get_value (adj) + delta_x * scroll_unit,
                              gtk_adjustment_get_lower (adj),
@@ -2301,7 +2304,10 @@

           adj = gtk_range_get_adjustment (GTK_RANGE (priv->vscrollbar));
           page_size = gtk_adjustment_get_page_size (adj);
-          scroll_unit = pow (page_size, 2.0 / 3.0);
+          if (event->direction == GDK_SCROLL_SMOOTH)
+            scroll_unit = 1.0;
+          else
+            scroll_unit = pow (page_size, 2.0 / 3.0);

           new_value = CLAMP (gtk_adjustment_get_value (adj) + delta_y * scroll_unit,
                              gtk_adjustment_get_lower (adj),
